### PR TITLE
Bugs/7593 - with test fix

### PR DESCRIPTION
**Pull request by bradhead**
Added authZ checks to Immz and Patient Service and tidyed up the service result to use Forbid on authZ failure and construct a new JsonResult()

### CODE REVIEW

File: Apps/Immunization/src/Controllers/ImmunizationController.cs [hunk 1]
```diff
@@ -35,27 +39,50 @@ public class ImmunizationController : ControllerBase
...
+            this.authorizationService = authorizationService;
         }
 
         /// <summary>
         /// Gets a json list of immunization records.
         /// </summary>
+        /// <param name="hdid">The health directed id for the subject.</param>
         /// <returns>a list of immunization records.</returns>
         /// <response code="200">Returns the List of Immunization records.</response>
         /// <response code="401">The client is not authorzied to retrieve the list.</response>
```
Comment (TiagoGraf89) [inline Apps/Immunization/src/Controllers/ImmunizationController.cs]: nit: 403 is missing

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_183)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

**Comment by sslaws** [comment]
Immunization tests are failing... please fix issue before merging.

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_183)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_183)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_183)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_183)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_183)

**Comment by bradhead** [comment]
> Imms tests are failing

the darn functional tests are old thinking.. hard-coded base Url, and the overall route wasn't thought out. I fixed the test for now... so we can assert that a 401 works now, but this needs to be looked at once we resume Immz. I just wanted to be sure we have Authorization protections on all of our services, even this one... updated this evening.

**Pull request merged**